feat(subscription): emit event on error sign in

StaberindeZA opened the pull request:
Because:

* The "Sign In" button in the "account exists" error message does not
  trigger the event fxa_pay_account_setup - other

This commit:

* Adds the onClick function to the Sign In button, which will trigger
  the event.

Closes #
[10838](https://github.com/mozilla/fxa/issues/10838)

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

vbudhram reviewed (approved):
@StaberindeZA Thanks!

Not sure if you knew this, but you can grep though the log events locally

```
pm2 logs payments | grep 'amplitudeEvent'
```

which gives 
```
fxa-payments-server.amplitudeEvent: {"op":"amplitudeEvent","event_type":"fxa_pay_account_setup - other","time":1636401567148,"user_id":"9c670712d0124346b1475fe3a79601a5","device_id":"13773a2b02d34857a72e10390149ad85","session_id":1636401533354,"app_version":"218.6","os_name":"Mac OS X","os_version":"10.15","event_properties":{"plan_id":"plan_GqM9N6qyhvxaVk","product_id":"prod_GqM9ToKK62qjkK","checkout_type":"without-account","other":"click-signnin"},"user_properties":{"flow_id":"b605bc7635e284ce5c69eb65f5bd606fe43b4ebe33755c09b5e36c404875b3ac","ua_browser":"Firefox","ua_version":"93.0","utm_campaign":"123done"}}
```

StaberindeZA commented:
@vbudhram awesome, thank you for the info!

I also updated the PR's subject the commit message subject. 

(Note that I updated the commit that got merged into `main` during Squash and Merge [30992d6](https://github.com/mozilla/fxa/commit/30992d60053560603c06db8f39b2d0a8a90c6af0). So the commit of this PR won't show the changes.)
